src/main/java/org/resitrack/controller/HouseController.java

```diff
@@ -1,4 +1,178 @@
 package org.resitrack.controller;
 
+import org.resitrack.entity.House;
+import org.resitrack.util.CommonUtil;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Create a House Controller.
+ *
+ * @author TuDucThinh
+ * @since 13/3/2023
+ */
 public class HouseController {
+    /**
+     * Used to access Scanner.
+     */
+    private final Scanner scanner;
+    /**
+     * Used to access townController.
+     */
+    private TownController townController;
+    /**
+     * Used to create a list of houses.
+     * Helps users to store houses that have just been saved to town
+     */
+    private List<House> houses;
+
+    public HouseController() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    public void setTownController(TownController townController) {
+        this.townController = townController;
+    }
+
+    public void setHouses(List<House> houses) {
+        this.houses = houses;
+    }
+
+    /**
+     * This function use for create a new house inside a town.
+     * After entering, you will be notified as follows: Successfully add the house
+     * If you want to add another house, press 1 if you want to exit, press 0
+     */
+    public void createNewHouse() {
+
+        String townId = getTownId();
+        String townName = townController.getTownById(townId).getName();
+        int numberOfHouse = 1;
+        int choice;
+        if (townId == null) {
+            System.out.println("Town is empty, could not create a new house. Please create a new town first");
+            return;
+        }
+
+        do {
+            String houseNumber = getHouseNumber(townId);
+
+            this.houses.add(createHouse(houseNumber, townId));
+            System.out.println("\nAdded " + numberOfHouse + " house. The latest home address is: " + houseNumber + " " + townName);
+            System.out.println();
+            System.out.print("Enter 0 to exit, Enter 1 to continue: ");
+            choice = scanner.nextInt();
+            scanner.nextLine();
+            if (choice == 1) {
+                numberOfHouse++;
+            }
```

Comment:
We can remove this check by make choice is String type. 

```
        String choice;
        do {
            System.out.println("Adding a new house");
            String houseNumber = getHouseNumber(townId);
            this.houses.add(createHouse(houseNumber, townId));

            System.out.println("\nAdded " + numberOfHouse + " house. The latest home address is: " + houseNumber + " " + townName);
            System.out.print(">Enter 0 to exit, or press any key to continue!");
            choice = scanner.nextLine();
            numberOfHouse++;

        } while (!choice.equalsIgnoreCase("0"));
        
  ```

Reply:
ok i will replace it now